flytectl/cmd/upgrade/upgrade.go

```diff
@@ -114,19 +114,20 @@ func upgrade(u *updater.Updater) (string, error) {
 }
 
 func isUpgradeSupported(goos platformutil.Platform) (bool, error) {
-	latest, err := github.FlytectlReleaseConfig.GetLatestVersion()
+	latest, err := github.FlytectlReleaseConfig.Provider.GetLatestVersion()
 	if err != nil {
 		return false, err
 	}
 
-	if isGreater, err := util.IsVersionGreaterThan(latest, stdlibversion.Version); err != nil {
+	compatibleVersion := strings.TrimPrefix(latest, fmt.Sprintf("%s/", github.FlytectlReleaseConfig.ExecutableName))
```

Comment:
Hi, since `GetLatestVersion` is used by the updater, so we can't move it there. A workaround is to define a new function that trims the prefix in the provider.